storm/src/main/storm/mesos/resources/RangeResource.java

```diff
@@ -0,0 +1,214 @@
+                                   resourceType, reservationType.toString(), rangeResourceEntry.toString(), toString(availableResourcesByReservationType.get(reservationType)));
+    throw new ResourceNotAvailabeException(message);
+  }
+
+  /**
+   * Remove/Reserve range from available ranges
+   * {@param rangeResourceEntry} range resource to removeAndGet
+   * {@param reservationTypeComparator} comparator like {@link storm.mesos.resources.DefaultReservationTypeComparator}
+   * to determine the priority of reservation types. When resources of several reservation types ({@link storm.mesos.resources.ReservationType})
+   * are available, the priority of reservation type removed is governed by {@link storm.mesos.resources.DefaultReservationTypeComparator}
+   */
+  @Override
+  public List<ResourceEntry> removeAndGet(RangeResourceEntry rangeResourceEntry, Comparator<ReservationType> reservationTypeCompartor) throws ResourceNotAvailabeException {
+    if (isAvailable(rangeResourceEntry)) {
+      List<ReservationType> reservationTypeList = Arrays.asList(ReservationType.values());
+      Collections.sort(reservationTypeList, reservationTypeCompartor);
+      return removeAndGet(reservationTypeList, rangeResourceEntry);
+    }
+
+    String message = String.format("ResourceType '%s' is not available. Requested value: %s Available: %s",
+                                   resourceType, rangeResourceEntry, toString());
+    throw new ResourceNotAvailabeException(message);
+  }
+
+  public String toString(List<RangeResourceEntry> ranges) {
+    List<String> rangeStrings = new ArrayList<>();
+    for (RangeResourceEntry range : ranges) {
+      String beginStr = String.valueOf(range.getBegin());
+      String endStr = String.valueOf(range.getEnd());
+      /*
+      * A Range representing a single number still has both Range.begin
+      * and Range.end populated, but they are set to the same value.
+      * In that case we just return "N" instead of "N-N".
+      */
+      if (range.getBegin() == range.getEnd()) {
+        rangeStrings.add(beginStr);
+      } else {
+        rangeStrings.add(String.format("%s-%s", beginStr, endStr));
+      }
+    }
+    return String.format("[%s]", StringUtils.join(rangeStrings, ","));
+  }
+
+  public String toString(ReservationType reservationType) {
+    return String.format("%s : %s", reservationType, toString(availableResourcesByReservationType.get(reservationType)));
+  }
+
+  public String toString() {
+    List<RangeResourceEntry> resourceRanges = new ArrayList<>();
+    for (Map.Entry<ReservationType, List<RangeResourceEntry>> entry : availableResourcesByReservationType.entrySet()) {
+      resourceRanges.addAll(entry.getValue());
+    }
+    return toString(resourceRanges);
+  }
+
+  private List<ResourceEntry> removeAndGet(Collection<ReservationType> reservationTypes, RangeResourceEntry rangeResourceEntry) {
```

Comment:
Can we move this method up to ~143, so that it follows all of the other `removeAndGet` method? Also can we add a comment to this method similar to the others?